### ISSUE TITLE
app_url with multistore and store_code in url setup

### ISSUE THREAD

**Issue by 6ui11em** [magewirephp/magewire#29]
We are getting a 404 error when using magewire in multistore with store code in the url.

For exemple, Magewire call this url:

https://magento2.docker/magewire/post/livewire/message/magewire.history.grid 

Then the URL is redirected to:

https://magento2.docker/de/magewire/post/livewire/message/magewire.history.grid 

And then we get the 404. This is because the redirect make a GET request and the controller expects a POST.

We fixed it with the following modifications:

```
diff --git a/vendor/magewirephp/magewire/src/ViewModel/Magewire.php b/vendor/magewirephp/magewire/src/ViewModel/Magewire.php
index 51a45a18..9e6ea54b 100644
--- a/vendor/magewirephp/magewire/src/ViewModel/Magewire.php
+++ b/vendor/magewirephp/magewire/src/ViewModel/Magewire.php
@@ -71,7 +71,7 @@ class Magewire implements ArgumentInterface
     public function getApplicationUrl(): string
     {
         try {
-            $base = $this->storeManager->getStore()->getBaseUrl(UrlInterface::URL_TYPE_WEB);
+            $base = $this->storeManager->getStore()->getBaseUrl();
             return $base . trim($this->getPostRoute(), '/');
         } catch (NoSuchEntityException $exception) {
             return $this->getPostRoute();
```

```
diff --git a/vendor/magewirephp/magewire-requirejs/src/view/base/templates/page/js/magewire-config.phtml b/vendor/magewirephp/magewire-requirejs/src/view/base/templates/page/js/magewire-config.phtml
index 7e099429..c1aec3aa 100644
--- a/vendor/magewirephp/magewire-requirejs/src/view/base/templates/page/js/magewire-config.phtml
+++ b/vendor/magewirephp/magewire-requirejs/src/view/base/templates/page/js/magewire-config.phtml
@@ -18,7 +18,7 @@ $magewireScripts = $block->getData('viewModel');
 {
     "*": {
         "Magewirephp_MagewireRequireJs/js/magewire": {
-            "app_url": "<?= $escaper->escapeUrl($magewireScripts->getPostRoute()) ?>",
+            "app_url": "<?= $escaper->escapeUrl($magewireScripts->getApplicationUrl()) ?>",
             "app_debug": <?= $escaper->escapeJs($magewireScripts->isDeveloperMode() ? 'true' : 'false') ?>
         }
     }

```

**Comment by wpoortman** [comment]
Hi @6ui11em,

I've did some testing and this should be fixed within the upcoming 1.6.0 release. I'll close the issue as soon as the release gets shipped.

I still need to work on the RequireJS module as soon as I have time. Please feel free to create a PR if it takes to long.

**Comment by 6ui11em** [comment]
Hi @wpoortman , Thank you for the information.
I will wait for the new release and create a PR to RequireJS if it's necessary.

**Comment by wpoortman** [comment]
I just released [1.6.0](https://twitter.com/wpoortman/status/1516699678817067008?s=20&t=rYn8AdD4EZ5frVUPJq87eg)
